File: gssg-interface/src/main/java/com/gssg/gssgbe/common/type/ProfileDogType.java
package com.gssg.gssgbe.common.type;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum ProfileDogType {

    BAEKGU("백구", false),
    JANGGUN("장군", false),
    WUYU("우유", false),
    YORK("요크", false),
    CORGI("코기", false),
    SILVER("실버", false),
    ALEX("알렉스", false),
    ;

    private final String name;
    private final Boolean highlight;

    public static ProfileDogType getDefault() {
        return ProfileDogType.BAEKGU;
    }
}

File: gssg-domain/src/main/kotlin/com/gssg/assets/domain/member/enums/ProfileDogType.kt
package com.gssg.assets.domain.member.enums

import com.gssg.assets.config.utils.notNull

/**
 * @Author Heli
 */
enum class ProfileDogType(
    val displayName: String,
    val highlight: Boolean
) {
    BAEKGU("백구", false),
    JANGGUN("장군", false),
    WUYU("우유", false),
    YORK("요크", false),
    CORGI("코기", false),
    SILVER("실버", false),
    ALEX("알렉스", false);

    companion object {
        private val types = values().associateBy { it.name.lowercase() }
        fun of(type: String) =
            types[type.lowercase()].notNull { "can not parse Member.ProfileDogType by $type" }
    }
}
